### DISCUSSION TITLE
Data discrepancy between published app environment and Budibase database

### DISCUSSION THREAD

**Discussion by amalzulkipli** [Budibase/budibase#14188] [Help] [answered]
Hello, I'm new to the Budibase service and currently using the Budibase database for my retail inventory system to track expiry dates.

After publishing my app, my co-workers began using it for data entry. Based on their feedback, I added some new features. However, before publishing these updates, I noticed that the data in the Budibase database doesn't match what's in the published environment.

Can I safely publish my updates? I've read that publishing can overwrite data. If it's not safe to publish, how can I synchronize the data from the published environment back to the Budibase database?

Any advice would be greatly appreciated. Thank you!

**Comment by ConorWebb96** [comment, marked as answer]
Hey @amalzulkipli,

Could this be related to the answer on this discussion here? https://github.com/Budibase/budibase/discussions/13267#discussioncomment-8839746

TLDR: Data is separated for published and development. If you want both development and published to be in sync you'll need to re-publish your app. I would recommend only editing records on your published app to avoid any out of syncness from occuring. Alternatively creating new records that you can then delete before publishing would be a good flow when making or testing in development.

The only way records will be removed is if you removed them in develop. Any changes your colleagues have made in the published version will be prioritised whenever you next publish your app.

I hope this makes sense.

If you want to ensure that everything is in sync. Looking into the public api and automations would be your best bet. What you could do. Setup an automation that will get all records via the pubic api loop over them and then amend the development database so everything is the same.
